tests/test_project.py

```diff
@@ -2491,13 +2495,34 @@ def test_project_schema_version_migration(self, implicit_version):
                     )
                 )
 
+            # Create a shell history file.
+            history_fn = os.path.join(dirname, ".signac_shell_history")
+            with open(history_fn, "w") as f:
+                f.write(
+                    textwrap.dedent(
+                        """\
+                        print(project)"""
+                    )
+                )
+
+            # Create a statepoint cache. Note that this cache does not
+            # correspond to actual statepoints since we don't currently have
+            # any in this project, but that's fine for migration testing.
+            history_fn = os.path.join(dirname, ".signac_sp_cache.json.gz")
+            sp = {"a": 1}
+            with gzip.open(history_fn, "wb") as f:
+                f.write(json.dumps({calc_id(sp): sp}).encode())
```

Comment:
This might be a better regression test if it did not re-use any internal machinery from the current signac implementation? Perhaps this only matters if we change our data schema more dramatically.
```suggestion
                f.write(b'{"42b7b4f2921788ea14dac5566e6f06d0": {"a": 1}}')
```

Reply:
We're not testing the contents of this file at all right now. We could make the test more stringent to validate that the file isn't corrupted during the migration, I just didn't go that far. I'm not sure that it's worth the effort right now.

Reply:
Okay. Maybe add a TODO, otherwise I agree we don’t need to invest more effort here.

Reply:
I think it's probably not worth worrying about. Worst case scenario, the user just has to run update-cache again.